lib/matplotlib/backend_managers.py

```diff
@@ -14,6 +18,188 @@
+        self._full_screen_flag = False
+
+        if is_interactive():
+            self.window.show()
+
+        def notify_axes_change(fig):
+            'this will be called whenever the current axes is changed'
+            if self.toolmanager is None and self.toolbar is not None:
+                self.toolbar.update()
+        self.figure.add_axobserver(notify_axes_change)
+
+    @property
+    def figure(self):
+        return self._figure
+
+    @figure.setter
+    def figure(self, figure):
+        if hasattr(self, '_figure'):
+            raise NotImplementedError
+
+        if not figure.canvas:
+            self._backend.FigureCanvas(figure, manager=self)
+        self._figure = figure
+
+    @property
+    def canvas(self):
+        return self._figure.canvas
+
+    def destroy(self, *args):
+        """Called to destroy this FigureManager.
+        """
+
+        # Make sure we run this routine only once for the FigureManager
+        # This ensures the nasty __del__ fix below works.
+        if getattr(self, '_destroying', False):
+            return
+
+        self._destroying = True
+        self.figure.canvas.destroy()
+        if self.toolbar:
+            self.toolbar.destroy()
+        self.window.destroy()
+
+        # Fix as for some reason we have extra references to this#
+        # i.e. ``del self._mainloop`` doesn't work
+        self._mainloop.__del__()
+
+        s = 'window_destroy_event'
+        event = FigureManagerEvent(s, self)
+        self._callbacks.process(s, event)
+
+    def show(self):
+        """Shows the figure"""
+        self.window.show()
+
+    def full_screen_toggle(self):
+        """Toggles whether we show fullscreen, alternatively call
+        `window.fullscreen()`"""
+        self._full_screen_flag = not self._full_screen_flag
+        self.window.set_fullscreen(self._full_screen_flag)
+
+    def resize(self, w, h):
+        """"For gui backends, resize the window (in pixels)."""
+        self.window.resize(w, h)
+
+    def get_window_title(self):
+        """
+        Get the title text of the window containing the figure.
+        Return None for non-GUI backends (e.g., a PS backend).
+        """
+        return self.window.get_window_title()
+
+    def set_window_title(self, title):
+        """
+        Set the title text of the window containing the figure.  Note that
+        this has no effect for non-GUI backends (e.g., a PS backend).
+        """
+        self.window.set_window_title(title)
+
+    @property
+    def backend(self):
+        return self._backend
+
+    def _get_toolbar(self):
+        # must be inited after the window, drawingArea and figure
+        # attrs are set
+        if rcParams['toolbar'] == 'toolmanager':
+            toolbar = self._backend.Toolbar(self.toolmanager)
+        else:
+            toolbar = None
+        return toolbar
+
+    def show_popup(self, msg):
+        """
+        Display message in a popup -- GUI only
+        """
+        pass
```

Comment:
Does anyone have any objection to getting rid of this method?  As far as I can tell we have never used it.  Also we will rcParam the new ``FigureManager`` so...